### PR TITLE
Feature login

### PR DESCRIPTION
**Pull request by startled-cat**

### CODE REVIEW

File: src/components/AccountForm/AccountForm.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  const [nameError, setNameError] = useState(false);
+
+  const reset = () => {
+    setPassword('');
+    setPhone(user.phoneNumber);
+    setName(user.firstName);
+  };
+
+  const handleChangePassword = () => {
+    let good: boolean = true;
```
Comment (Luvkitri) [inline src/components/AccountForm/AccountForm.tsx]: Typing is unnecessary when the variable is being initialised with value.

File: src/components/AccountForm/AccountForm.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    } else {
+      setPasswordError(false);
+    }
+    if (good) {
+      onPasswordChange(passwordValue);
+      setPassword('');
+    }
+  };
+  const handleButtonClick = () => {
+    let good: boolean = true;
```
Comment (Luvkitri) [inline src/components/AccountForm/AccountForm.tsx]: Same as above.

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -1,15 +1,65 @@
...
+
+  const handleUsernameButtonClick = () => {
+    navigate('/account');
+  };
+  const handleSignOutButtonClick = () => {
+    signOut();
+    navigate('/');
+  };
+  let buttons: JSX.Element = <b></b>;
+  const buttonsSignedIn = (
```
Comment (Luvkitri) [inline src/components/Header/Header.tsx]: You can make those as functions: renderSignedInButtons() and renderAnonymousButtons() and in return:
`
{isSignedIn ? renderSignedInButtons() : renderAnonymousButtons()}
`

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -1,15 +1,65 @@
 import { Button, Typography } from '@mui/material';
 import * as Styled from './Header.styles';
+import { useNavigate } from 'react-router-dom';
+import { isSignedIn, signOut, getUserEmail, onSignChange } from '@utils/login';
+import { render } from 'react-dom';
+import { useState } from 'react';
 
 const Header = () => {
+  const navigate = useNavigate();
+  const [signedIn, setSignedIn] = useState(isSignedIn());
```
Comment (Luvkitri) [inline src/components/Header/Header.tsx]: Make it isSignedIn - for booleans always try to start with 'is' - it's more readable that way.

File: src/components/LoginForm/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+const LoginForm: FC<LoginProps> = ({ onLogin }) => {
+  const [emailValue, setEmail] = useState('');
+  const [passwordValue, setPassword] = useState('');
+
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+
+  const handleButtonClick = () => {
+    let good: boolean = true;
```
Comment (Luvkitri) [inline src/components/LoginForm/LoginForm.tsx]: Same here, boolean is not needed.

File: src/components/RegisterForm/RegisterForm.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  const [phoneValue, setPhone] = useState('');
+  const [nameValue, setName] = useState('');
+
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [phoneError, setPhoneError] = useState(false);
+  const [nameError, setNameError] = useState(false);
+
+  const handleButtonClick = () => {
+    let good: boolean = true;
```
Comment (Luvkitri) [inline src/components/RegisterForm/RegisterForm.tsx]: Same as above

File: src/pages/Account/Account.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        console.log({ response });
+        alert('Changes saved.');
+      })
+      .catch((error) => {
+        alert(error);
+      });
+  };
+
+  let form;
+  if (user) {
```
Comment (Luvkitri) [inline src/pages/Account/Account.tsx]: I would make it as:
`
return <Styled.Wrapper>{user && <AccountForm .. />}</ Styled.Wrapper>
`

File: src/pages/Login/Login.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import * as Styled from '@layout/Layout.styles';
+import { signIn } from '@utils/login';
+
+const Login: FC = () => {
+  const navigate = useNavigate();
+  const handleLoginButtonClick = async (
+    emailValue: string,
+    passwordValue: string,
+  ) => {
+    if (await signIn(emailValue, passwordValue)) {
```
Comment (Luvkitri) [inline src/pages/Login/Login.tsx:14]: try...catch in this scenario?

File: src/utils/login.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+export const signIn = async (
+    emailValue: string,
+    passwordValue: string,
+) => {
+    let token: string;
+    try {
+        token = await User.login(emailValue, passwordValue);
+        console.log({ token });
+    } catch (error) {
+        return false
```
Comment (Luvkitri) [inline src/utils/login.ts:30]: Now I see the reason behind those if's in those cases throwing an Error is better practice. For this implementation it doesn't matter, so you can leave it be

**Review by Luvkitri — CHANGES_REQUESTED** [review]
Some small adjustments, but overall we can go with that. Once I'm done with accommodations I'll try to add some more usability.

File: src/components/AccountForm/AccountForm.tsx (hunk 1 above)
Comment (startled-cat) [reply src/components/AccountForm/AccountForm.tsx]: Addressed in commit: [03298211be1510af94c17aaecbe74406efdebccf](https://github.com/Luvkitri/zpab-frontend/pull/2/commits/03298211be1510af94c17aaecbe74406efdebccf)

File: src/components/AccountForm/AccountForm.tsx (hunk 2 above)
Comment (startled-cat) [reply src/components/AccountForm/AccountForm.tsx]: Addressed in commit: [03298211be1510af94c17aaecbe74406efdebccf](https://github.com/Luvkitri/zpab-frontend/pull/2/commits/03298211be1510af94c17aaecbe74406efdebccf)

File: src/components/Header/Header.tsx (hunk 3 above)
Comment (startled-cat) [reply src/components/Header/Header.tsx]: Addressed in commit: [03298211be1510af94c17aaecbe74406efdebccf](https://github.com/Luvkitri/zpab-frontend/pull/2/commits/03298211be1510af94c17aaecbe74406efdebccf)

File: src/components/Header/Header.tsx (hunk 4 above)
Comment (startled-cat) [reply src/components/Header/Header.tsx]: Addressed in commit: [03298211be1510af94c17aaecbe74406efdebccf](https://github.com/Luvkitri/zpab-frontend/pull/2/commits/03298211be1510af94c17aaecbe74406efdebccf)

File: src/components/LoginForm/LoginForm.tsx (hunk 5 above)
Comment (startled-cat) [reply src/components/LoginForm/LoginForm.tsx]: Addressed in commit: [03298211be1510af94c17aaecbe74406efdebccf](https://github.com/Luvkitri/zpab-frontend/pull/2/commits/03298211be1510af94c17aaecbe74406efdebccf)

File: src/components/RegisterForm/RegisterForm.tsx (hunk 6 above)
Comment (startled-cat) [reply src/components/RegisterForm/RegisterForm.tsx]: Addressed in commit: [03298211be1510af94c17aaecbe74406efdebccf](https://github.com/Luvkitri/zpab-frontend/pull/2/commits/03298211be1510af94c17aaecbe74406efdebccf)

File: src/pages/Account/Account.tsx (hunk 7 above)
Comment (startled-cat) [reply src/pages/Account/Account.tsx]: Addressed in commit: [03298211be1510af94c17aaecbe74406efdebccf](https://github.com/Luvkitri/zpab-frontend/pull/2/commits/03298211be1510af94c17aaecbe74406efdebccf)

**Review by Luvkitri — APPROVED** [review]

**Pull request merged**